integration/map_test.go

```diff
@@ -31,59 +31,138 @@ import (
 }
 
-func TestInclusionWithEnv(t *testing.T) {
+func TestInclusion(t *testing.T) {
 	ctx := context.Background()
-	env, err := integration.NewMapEnv(ctx, "TestInclusionWithEnv")
+	env, err := integration.NewMapEnv(ctx, "TestInclusion")
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer env.Close()
 	for _, tc := range []struct {
 		desc string
 		trillian.HashStrategy
-		index, value []byte
+		iv [][]byte
 	}{
 		{
-			desc:         "maphasher",
+			desc:         "maphasher single",
+			HashStrategy: trillian.HashStrategy_TEST_MAP_HASHER,
+			iv: [][]byte{
+				testonly.TransparentHash("A"), []byte("A"),
+			},
+		},
+		{
+			desc:         "maphasher multi",
 			HashStrategy: trillian.HashStrategy_TEST_MAP_HASHER,
-			index:        testonly.TransparentHash("A"),
-			value:        []byte("A"),
+			iv: [][]byte{
+				testonly.TransparentHash("A"), []byte("A"),
+				testonly.TransparentHash("B"), []byte("B"),
+				testonly.TransparentHash("C"), []byte("C"),
+			},
 		},
 	} {
-		treeParams := stestonly.MapTree
-		treeParams.HashStrategy = tc.HashStrategy
-		tree, err := env.AdminClient.CreateTree(ctx, &trillian.CreateTreeRequest{
-			Tree: treeParams,
-		})
+		tree, hasher, err := newTreeWithHasher(ctx, env, tc.HashStrategy)
 		if err != nil {
-			t.Errorf("%v: CreateTree(): %v", tc.desc, err)
+			t.Errorf("%v: newTreeWithHasher(%v): %v", tc.desc, tc.HashStrategy, err)
+		}
+		mapID := tree.TreeId
+		client := env.MapClient
+		leaves := createMapLeaves(tc.iv...)
+
+		if _, err := client.SetLeaves(ctx, &trillian.SetMapLeavesRequest{
+			MapId:  mapID,
+			Leaves: leaves,
+		}); err != nil {
+			t.Errorf("%v: SetLeaves(): %v", tc.desc, err)
 			continue
 		}
 
-		mapID := tree.TreeId
-		hasher, err := hashers.NewMapHasher(tree.HashStrategy)
+		indexes := [][]byte{}
+		for _, l := range leaves {
+			indexes = append(indexes, l.Index)
+		}
+		getResp, err := client.GetLeaves(ctx, &trillian.GetMapLeavesRequest{
+			MapId:    mapID,
+			Index:    indexes,
+			Revision: -1,
+		})
 		if err != nil {
-			t.Errorf("%v: NewMapHasher(): %v", tc.desc, err)
+			t.Errorf("%v: GetLeaves(): %v", tc.desc, err)
+			continue
+		}
+
+		rootHash := getResp.GetMapRoot().GetRootHash()
+		for _, m := range getResp.GetMapLeafInclusion() {
+			index := m.GetLeaf().GetIndex()
+			leafHash := m.GetLeaf().GetLeafHash()
+			proof := m.GetInclusion()
+			if err := merkle.VerifyMapInclusionProof(mapID, index,
+				leafHash, rootHash, proof, hasher); err != nil {
+				t.Errorf("%v: VerifyMapInclusionProof(): %v", tc.desc, err)
+			}
+		}
+	}
+}
+
+func TestInclusionLarge(t *testing.T) {
+	t.Skip("This test causes db connection starvation: Too many connections")
+	// Don't run tests with the larger numbers of leaves in short mode.
+	ShortCutoff := 150
+
+	ctx := context.Background()
+	env, err := integration.NewMapEnv(ctx, "TestInclusionLarge")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer env.Close()
+	for _, tc := range []struct {
+		desc string
+		trillian.HashStrategy
```

Comment:
Also it's a bit funny with the mix of private and exported naming schemes in here, is there a reason for that?